process.go

```diff
@@ -0,0 +1,84 @@
+package containerd
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"syscall"
+
+	"github.com/containerd/containerd/api/services/execution"
+	taskapi "github.com/containerd/containerd/api/types/task"
+	protobuf "github.com/gogo/protobuf/types"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type process struct {
+	task *task
+
+	pidLock sync.Mutex
+
+	io   *IO
+	pid  uint32
+	spec *specs.Process
+}
+
+func (p *process) Pid() uint32 {
+	return p.pid
```

Comment:
This will just return 0 when its not set, just need to be documented that the pid is not set until Start returns.  There is not much we can do without `runc exec` having a create/start